internal/operator/zitadel/kinds/backups/bucket/adapt.go

```diff
@@ -88,47 +88,60 @@ func AdaptFunc(
 
 		queryM, destroyM, checkMigrationDone, cleanupMigration, err := migration.AdaptFunc(monitor, namespace, "restore", labels, secretPasswordName, migrationUser, users, nodeselector, tolerations)
 
+		destroyers := make([]zitadel.DestroyFunc, 0)
 		destroyS, err := secret.AdaptFuncToDestroy(namespace, secretName)
 		if err != nil {
 			return nil, nil, nil, err
 		}
 
-		queryS, err := secret.AdaptFuncToEnsure(namespace, secretName, labels, map[string]string{secretKey: desiredKind.Spec.ServiceAccountJSON.Value})
-		if err != nil {
-			return nil, nil, nil, err
-		}
-
-		queriers := make([]zitadel.QueryFunc, 0)
-		destroyers := make([]zitadel.DestroyFunc, 0)
 		for _, feature := range features {
 			switch feature {
 			case "backup", "instantbackup":
-				queriers = append(queriers,
-					zitadel.ResourceQueryToZitadelQuery(queryS),
-					queryB,
-				)
 				destroyers = append(destroyers,
 					zitadel.ResourceDestroyToZitadelDestroy(destroyS),
 					destroyB,
 				)
 			case "restore":
-				queriers = append(queriers,
-					queryC,
-					zitadel.EnsureFuncToQueryFunc(checkAndCleanupC),
-					queryM,
-					zitadel.EnsureFuncToQueryFunc(checkMigrationDone),
-					zitadel.EnsureFuncToQueryFunc(cleanupMigration),
-					queryR,
-					zitadel.EnsureFuncToQueryFunc(checkAndCleanupR),
-				)
 				destroyers = append(destroyers,
+					zitadel.ResourceDestroyToZitadelDestroy(destroyS),
 					destroyC,
 					destroyM,
 					destroyR,
 				)
 			}
 		}
 
+		queriers := make([]zitadel.QueryFunc, 0)
+		if desiredKind.Spec != nil && desiredKind.Spec.ServiceAccountJSON != nil {
+			queryS, err := secret.AdaptFuncToEnsure(namespace, secretName, labels, map[string]string{secretKey: desiredKind.Spec.ServiceAccountJSON.Value})
+			if err != nil {
+				return nil, nil, nil, err
+			}
+
+			for _, feature := range features {
+				switch feature {
```

Comment:
Shouldn't we return an error if spec isn't nil but SAJSON is?

Reply:
Not in my opinion, as this correction was necessary since the SAJSON should be allowed to be nil. Otherwise even the definition of a backup without a SAJSON, which you have to define with "writesecret", will be seen as an error. In the same way you can't even define the secret, as the error is in the adapt step, which is also used by all commands.

Reply:
So, we could check for the SAJSON in the query phase then?

Reply:
Still think not having a SAJSON in the yml shouldn't be an error, as the current reaction is that the secret gets written only if the attribute is filled. If you want you can change it, but I see no positives about it.